contribute/dev_setup.rst

```diff
@@ -0,0 +1,100 @@
+
+#. GitHub user account (for the rest of this document we will call our user *Prasad*)
+#. Git installed
+
+This document doesn't cover the process of satisfying these prerequisites, it is left as an exercise for you (Google is your friend).  The remainder of this document assumes that these prerequisites have been met.  This document covers setup from the command line if you are using a GUI like 'GitHub for Windows' or 'Pycharm' then you will need to adjust the commands for the GUI you are using.
+
+The goal here is to get a working copy of source code, tests, and documentation onto your PC so that you can begin development.  To make this happen you will need to fork a :term:`prime OpenCMISS repository`, make a clone onto your PC, and set up the Git remotes.  In :numref:`fig_devSetup_githubRepos` you can see a pictorial representation of what we are aiming to achieve.
+
+.. _fig_devSetup_githubRepos:
+
+.. figure:: images/OpenCMISSProcesses-GitHubRepos.png
+   :align: center
+   :alt: Setup of Git repositories
+
+   Setup of repositories for development
+
+The four steps to getting set up are detailed below.
+
+Forking your own copy
+=====================
+
+Login to GitHub using your credentials and go to the OpenCMISS GitHub organisation page::
+
+   https://github.com/OpenCMISS
+
+The OpenCMISS organisation contains the git repositories for each component of OpenCMISS:
+
+   - https://github.com/OpenCMISS/iron
+   - https://github.com/OpenCMISS/zinc
+   - https://github.com/OpenCMISS/examples
+   - https://github.com/OpenCMISS/documentation
+   - https://github.com/OpenCMISS/website
+   - https://github.com/OpenCMISS/manage
+
+Select the repository that you wish to develop and use the fork button to create a fork of the repository under your own account, see :numref:`fig_devSetup_githubFork` for locating this button. The remainder of this documentation will assume you have selected the iron repository, however, the steps outlined below can be applied to any of the repositories in the OpenCMISS organisation.
+
+.. _fig_devSetup_githubFork:
+
+.. figure:: images/OpenCMISSProcesses-GitHubForkButton.png
+   :align: center
+   :alt: Fork button of the OpenCMISS iron repository
+
+   Fork button for the OpenCMISS iron repository
+
+Clone
+=====
+
+You now need to clone the selected repository to your PC.  Do this by going to your fork (in this example user *prasad*'s fork) at::
+
+   https://github.com/prasad/iron
+
+.. warning::
+
+   Don't try and clone this location. Substitute your GitHub username for *prasad*.
+
+On the right hand side of the webpage on your fork of the repository you can get the link for cloning the repository to your PC, in our example::
+
+        git@github.com:prasad/iron.git
+
+Now clone the repository::
+
+        cd <somewhere/you/keep/development/code>
+        git clone git@github.com:prasad/iron.git
```

Comment:
We should keep our story consistent with setup: recommend making a top opencmiss folder (I call mine oc to keep paths short) and in it add a sub folder src, and clone your fork of iron there. This is where the setup script will clone iron etc. (admittedly it clones the prime repo by default, but you can switch to your own remote).
Did you know if you subsequently run setup or manage on the same opencmiss (or 'oc') folder it will discover and keep your sources for any components in the src folder? It's very convenient for rebuilding from scratch. This is the convenient way to get your own forks.